cmake/Modules/OpmStaticTargets.cmake

```diff
@@ -7,15 +7,23 @@
 
 # Macros
 function(opm_from_git repo name revision)
+  if(ARGN)
+    set(COMMANDS BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> --target ${ARGN}
+                 INSTALL_COMMAND)
```

Comment:
in principle one alternative would be to just build straight out of the source tree but then cmake bugs bites. it cannot handle a build directory which is a subdir of the source tree (it tries to copy and then fails since you change the source dir when the copy happens causing recursions). thus, the approach of doing another clone of your local git is necessary.